src/identity/main.cairo

```diff
@@ -88,17 +174,164 @@ mod Identity {
+    //
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        // todo: move these functions into a storage contract when components are available
+        fn get_extended(
+            self: @ContractState,
+            fn_name: felt252,
+            params: Span<felt252>,
+            offset: u8,
+            length: felt252,
+            domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                if length == offset.into() {
+                    break ();
+                }
+                let value = self._get(domain, base, offset);
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn get_unbounded(
+            self: @ContractState, fn_name: felt252, params: Span<felt252>, offset: u8, domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                let value = self._get(domain, base, offset);
+                if value == 0 {
+                    break ();
+                }
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn _get(
+            self: @ContractState, domain: u32, base: starknet::StorageBaseAddress, offset: u8,
+        ) -> felt252 {
+            starknet::storage_read_syscall(
+                domain, starknet::storage_address_from_base_and_offset(base, offset)
+            )
+                .unwrap_syscall()
+        }
+
+        fn set(
+            ref self: ContractState,
+            fn_name: felt252,
+            params: Span<felt252>,
+            offset: u8,
+            value: Span<felt252>,
+            domain: u32,
+        ) {
+            let base = self.compute_base_address(fn_name, params);
+            self._set(domain, base, value, offset: offset);
+        }
+
+        fn _set(
+            ref self: ContractState,
+            domain: u32,
+            base: starknet::StorageBaseAddress,
+            mut value: Span<felt252>,
+            offset: u8
+        ) {
+            match value.pop_front() {
+                Option::Some(v) => {
+                    starknet::storage_write_syscall(
+                        domain, starknet::storage_address_from_base_and_offset(base, offset), *v
+                    );
+                    self._set(domain, base, value, offset + 1);
+                },
+                Option::None(_) => {},
+            }
+        }
+
+        fn compute_base_address(
+            self: @ContractState, fn_name: felt252, params: Span<felt252>
+        ) -> StorageBaseAddress {
+            let mut _params = params;
+            let mut tmp: felt252 = hash::LegacyHash::hash(
```

Comment:
Could you add tests for this function in `test_addresses_comp.cairo`? I think this will fail for functions of 0 params. I might be wrong but I also feel like this is not exactly the same algorithm used by `.addr()` in Cairo 0. Try to compare it to ``utils::get_storage_var_address``outputs from starknet-rs.